fluent-bit-kinesis.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//  http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+    "C"
+    "unsafe"
+
+    "github.com/hossain-rayhan/amazon-kinesis-streams-for-fluent-bit/kinesis"
+    "github.com/aws/amazon-kinesis-firehose-for-fluent-bit/plugins"
+    "github.com/fluent/fluent-bit-go/output"
+
+    "github.com/sirupsen/logrus"
+)
+import (
+    "fmt"
+)
+
+var (
+    pluginInstances []*kinesis.OutputPlugin
+)
+
+func addPluginInstance(ctx unsafe.Pointer) error {
+    pluginID := len(pluginInstances)
+    output.FLBPluginSetContext(ctx, pluginID)
+    instance, err := newKinesisOutput(ctx, pluginID)
+    if err != nil {
+        return err
+    }
+
+    pluginInstances = append(pluginInstances, instance)
+    return nil
+}
+
+func getPluginInstance(ctx unsafe.Pointer) *kinesis.OutputPlugin {
+    pluginID := output.FLBPluginGetContext(ctx).(int)
+    return pluginInstances[pluginID]
+}
+
+
+func newKinesisOutput(ctx unsafe.Pointer, pluginID int) (*kinesis.OutputPlugin, error) {
+    stream := output.FLBPluginConfigKey(ctx, "stream")
+    logrus.Infof("[kinesis %d] plugin parameter stream = '%s'", pluginID, stream)
+    region := output.FLBPluginConfigKey(ctx, "region")
+    logrus.Infof("[kinesis %d] plugin parameter region = '%s'", pluginID, region)
+    dataKeys := output.FLBPluginConfigKey(ctx, "data_keys")
+    logrus.Infof("[kinesis %d] plugin parameter data_keys = '%s'", pluginID, dataKeys)
+    partitionKey := output.FLBPluginConfigKey(ctx, "partition_key")
+    logrus.Infof("[kinesis %d] plugin parameter partition_key = %s", pluginID, partitionKey)
+    roleARN := output.FLBPluginConfigKey(ctx, "role_arn")
+    logrus.Infof("[kinesis %d] plugin parameter role_arn = '%s'", pluginID, roleARN)
+    endpoint := output.FLBPluginConfigKey(ctx, "endpoint")
+    logrus.Infof("[kinesis %d] plugin parameter endpoint = '%s'", pluginID, endpoint)
+
+    if stream == "" || region == "" {
+        return nil, fmt.Errorf("[kinesis %d] stream and region are required configuration parameters", pluginID)
+    }
+    if partitionKey == "" {
+        logrus.Infof("[kinesis %d] no partition key provided! random one will be generated", pluginID)
```

Comment:
"random one will be generated" => "A random one will generated for each record"

Reply:
"A random one will generated for each record" or "A random one will be generated for each record"?